### ISSUE TITLE
`SnappyCodec.compress` should accept `Encoder` instance

### ISSUE THREAD

**Issue by nicholas-ramsey** [declerckt/kafkajs-snappy-typescript#1]
I'm experiencing the following error when publishing messages:
```
Error: Value is not either String or Buffer
    at Object.<anonymous> (/home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs-snappy-typescript@1.0.1/node_modules/kafkajs-snappy-typescript/dist/codec/SnappyCodec.js:31:42)
    at Generator.next (<anonymous>)
    at /home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs-snappy-typescript@1.0.1/node_modules/kafkajs-snappy-typescript/dist/codec/SnappyCodec.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs-snappy-typescript@1.0.1/node_modules/kafkajs-snappy-typescript/dist/codec/SnappyCodec.js:4:12)
    at Object.compress (/home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs-snappy-typescript@1.0.1/node_modules/kafkajs-snappy-typescript/dist/codec/SnappyCodec.js:30:16)
    at compressRecords (/home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs@2.2.4/node_modules/kafkajs/src/protocol/recordBatch/v0/index.js:87:16)
    at RecordBatch (/home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs@2.2.4/node_modules/kafkajs/src/protocol/recordBatch/v0/index.js:65:37)
    at /home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs@2.2.4/node_modules/kafkajs/src/protocol/requests/produce/v3/request.js:103:29
    at /home/logram/src/saferplates.io/node_modules/.pnpm/kafkajs@2.2.4/node_modules/kafkajs/src/protocol/requests/produce/v3/request.js:70:13 {
  code: 'InvalidArg'
}
```

Here's some code to reproduce the issue. I'm using `kafkajs@2.2.4` and `kafkajs-snappy-typescript@1.0.1`.
```ts
import { Kafka, CompressionCodecs, CompressionTypes } from "kafkajs"
import { SnappyCodec } from "kafkajs-snappy-typescript";

CompressionCodecs[CompressionTypes.Snappy] = new SnappyCodec().codec;

const kafka = new Kafka({
  clientId: "client-id",
  brokers: [],
})

const kafkaProducer = kafka.producer()

(async () => {
  await kafkaProducer.connect()
  await kafkaProducer.send({
      topic: "test-topic",
      messages: [{
          value: "test"
      }],
      acks: -1,
      compression: CompressionTypes.Snappy,
  })
})();
```

It looks like `compress` is meant to accept a kafkajs `Encoder` instance as an argument, and not a `string`, as described in the [_Producing_](https://github.com/tulios/kafkajs/blob/ff3b1117f316d527ae170b550bc0f772614338e9/docs/Producing.md#-other) documentation.

**Comment by RoverDaniell** [comment]
It looks like the latest released package 1.0.2 does not have this fix, it still has:
```
compress(msg) {
        return __awaiter(this, void 0, void 0, function* () {
            return (0, snappy_1.compress)(msg);
        });
    }
```

**Comment by declerckt** [comment]
Hi @RoverDaniell thank you for telling me, sorry I did not published correctly the 1.0.2 package, I bundled everything again to the [1.0.3](https://www.npmjs.com/package/kafkajs-snappy-typescript/v/1.0.3), you can check it seems fine now. If not tell me I will have a deeper look 😉 

```javascript
compress(encoder) {
    return __awaiter(this, void 0, void 0, function* () {
        return (0, snappy_1.compress)(encoder.buffer);
    });
}
```

Have a nice day !
